serving-api/src/main/scala/com/stratio/sparkta/serving/api/actor/ClusterLauncherActor.scala

```diff
@@ -0,0 +1,229 @@
+/**
+ * Copyright (C) 2016 Stratio (http://stratio.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.stratio.sparkta.serving.api.actor
+
+import java.io.File
+import scala.collection.JavaConversions._
+import scala.concurrent.duration._
+import scala.io.Source
+import scala.util.Failure
+import scala.util.Properties
+import scala.util.Success
+import scala.util.Try
+
+import akka.actor.Actor
+import akka.actor.ActorRef
+import akka.event.slf4j.SLF4JLogging
+import akka.pattern.ask
+import akka.util.Timeout
+import com.google.common.io.BaseEncoding
+import com.typesafe.config.Config
+import com.typesafe.config.ConfigRenderOptions
+import org.apache.spark.launcher.SparkLauncher
+
+import com.stratio.sparkta.driver.repository.ClusterConfigRepositoryComponent
+import com.stratio.sparkta.driver.service.StreamingContextService
+import com.stratio.sparkta.driver.util.HdfsUtils
+import com.stratio.sparkta.driver.util.PolicyUtils
+import com.stratio.sparkta.serving.api.actor.SparkStreamingContextActor._
+import com.stratio.sparkta.serving.core.SparktaConfig
+import com.stratio.sparkta.serving.core.constants.AppConstant
+import com.stratio.sparkta.serving.core.helpers.JarsHelper
+import com.stratio.sparkta.serving.core.models.AggregationPoliciesModel
+import com.stratio.sparkta.serving.core.models.PolicyStatusModel
+import com.stratio.sparkta.serving.core.models.SparktaSerializer
+import com.stratio.sparkta.serving.core.policy.status.PolicyStatusActor.Update
+import com.stratio.sparkta.serving.core.policy.status.PolicyStatusEnum
+
+class ClusterLauncherActor(policy: AggregationPoliciesModel,
+                           streamingContextService: StreamingContextService,
+                           clusterConfig: Config,
+                           hdfsConfig: Config,
+                           zookeeperConfig: Config,
+                           detailConfig: Config,
```

Comment:
has a lot of config arguments, maybe you should to think in dataclump -> http://martinfowler.com/bliki/DataClump.html